### PR TITLE
[codex] Add Grok-1 expert atlas analysis

### PR DESCRIPTION
**Pull request by rmems**
## What changed
- add a new `src/experts` analysis module that derives expert structure from `ModelInventory`
- add an `experts` CLI subcommand that builds an expert atlas and exports JSON or Markdown
- add expert-atlas schema types and report renderers for block-to-expert mapping, per-expert tensor associations, naming checks, and anomaly reporting

## Why
The repo could inventory Grok-1 tensors, but it could not yet answer how a checkpoint is organized at the expert level. This adds a parser-driven, structural MoE view without introducing runtime or inference logic.

## Impact
- `cargo run -- experts <path-to-checkpoint>` now produces a useful expert atlas
- downstream consumers can use the exported JSON and Markdown artifacts to inspect expert counts, slot patterns, and irregularities
- the change stays within the repo's analysis boundary: no forward pass, no logits, no quantization work

## Validation
- `cargo test`
- `cargo run -- experts /home/raulmc/Downloads/SNN_Quantization/grok-1-official/ckpt-0 --limit 14`

## Notes
- local `gh` auth is invalid on this machine, so the PR was opened via the GitHub connector
- a full-checkpoint atlas sweep was started locally; the smoke validation above completed successfully against the real Grok-1 checkpoint

### CODE REVIEW

**Comment by rmems** [comment]
@codeant-ai Can you make any adjustments please?

**Comment by rmems** [comment]
@codeant-ai yes please go ahead do it
